Add CodeQL analysis workflow

robertbrignull opened the pull request:
Slight disclaimer, I work at GitHub in the code scanning team, however I've also read Homestuck and I think this project to preserve it is awesome.

This PR adds a workflow that runs CodeQL analysis to find security problems and other programming errors. It's a pretty good tool for finding code problems and stopping them from being introduced in new PRs. Entirely up to you if you wantto use it. You can read a bit more at https://docs.github.com/en/code-security/secure-coding/about-code-scanning.

Unfortunately due to this being the first analysis and being from a fork you may not be able to see the list of alerts before merging but I once it is merged the workflow will run nightly as well as on new PRs and will tell you how many alerts are introduced or fixed by the PR. I have run the analysis myself so I can tell you the list of things it has found if you like.

robertbrignull commented on .github/workflows/codeql-analysis.yml:
```diff
@@ -0,0 +1,28 @@
+      uses: github/codeql-action/init@v1
+      with:
+        languages: 'javascript'
+        queries: security-and-quality
```
I've added `security-and-quality` here so it'll run more than just security queries and will find general programming errors as well as maintainability and reliability issues. You can read more at https://docs.github.com/en/code-security/secure-coding/configuring-code-scanning#running-additional-queries

robertbrignull commented:
Seems you should actually be able to view the alerts in this commit at https://github.com/Bambosh/unofficial-homestuck-collection/security/code-scanning?query=ref%3Arefs%2Fpull%2F87%2Fmerge+tool%3ACodeQL (This link is only visible to the project maintainer. I can't actually see it.)

GiovanH commented:
This looks very nice. Can you put together a workflow that runs on `develop` (like https://github.com/Bambosh/unofficial-homestuck-collection/blob/main/.github/workflows/main.yml does) and make a PR on the develop branch?

robertbrignull commented:
Sure. I've changed the workflow to target the develop branch and PRs against that branch, and I've changed this PR to targt the develop branch.

The above should be fine to avoid introducing alerts in PRs, and viewing the set of alerts on the develop branch. Probably worth nothing that the security tab by default shows you results for the default branch, so you'd have to make sure to select the branch you want. You may get a better experience running the workflow against main and develop, assuming there's not really any extra cost of doing so. But also that's something that can be changed later on so it's no big deal.
